forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/SyncFormDataPipelineListener.java

```diff
@@ -130,52 +131,52 @@ private void readInnerDataElements(int i, JsonNode jsonData, Set<FormElement> el
-
-        Mono<ResponseEntity<String>> entityMono = webClient.method(HttpMethod.POST).uri(getSyncApplicationUrl())
-                .accept(MediaType.APPLICATION_JSON).header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
-                .body(Mono.just(payload), String.class).retrieve().toEntity(String.class);
-
-        ResponseEntity<String> response = entityMono.block();
-        return response.getBody();
-    }
-
-    /**
-     * Prepares and returns the Form Data Elements.
-     *
-     * @param execution
-     * @return
-     */
-    private JsonNode prepareSyncData(DelegateExecution execution) throws IOException {
-        String submission = formSubmissionService.readSubmission(getUrl(execution));
-        if (submission.isEmpty()) {
-            throw new RuntimeException("Unable to retrieve submission");
-        }
-        JsonNode dataNode = new ObjectMapper().readTree(submission);
-        JsonNode dataElements = dataNode.findValue("data");
-        return dataElements;
-    }
-
-    private String getUrl(DelegateExecution execution) {
-        return String.valueOf(execution.getVariables().get("formUrl"));
-    }
-
-    /**
-     * Returns the endpoint of Sync API.
-     *
-     * @return
-     */
-    private String getSyncApplicationUrl() {
-        return this.reportUrl + "/sync-form-data";
-    }
+	/**
+	 * This method invokes Sync API and returns the response.
+	 *
+	 * @param execution
+	 * @return
+	 */
+	private String invokeSyncApplicationService(DelegateExecution execution) throws IOException {
+		Object dataJson = prepareSyncData(execution);
+		String payload = dataJson != null ? new ObjectMapper().writeValueAsString(dataJson) : null;
+		payload = (dataJson == null) ? new JsonObject().toString() : payload;
+
+		Mono<ResponseEntity<String>> entityMono = webClient.method(HttpMethod.POST).uri(getSyncApplicationUrl())
+				.accept(MediaType.APPLICATION_JSON).header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+				.body(Mono.just(payload), String.class).retrieve().toEntity(String.class);
+
+		ResponseEntity<String> response = entityMono.block();
+		return response.getBody();
+	}
+
+	/**
+	 * Prepares and returns the Form Data Elements.
+	 *
+	 * @param execution
+	 * @return
+	 */
+	private JsonNode prepareSyncData(DelegateExecution execution) throws IOException {
+		String submission = formSubmissionService.readSubmission(getUrl(execution));
+		if (submission.isEmpty()) {
+			throw new RuntimeException("Unable to retrieve submission");
+		}
+		JsonNode dataNode = new ObjectMapper().readTree(submission);
+		JsonNode dataElements = dataNode.findValue("data");
+		return dataElements;
+	}
+
+	private String getUrl(DelegateExecution execution) {
+		return String.valueOf(execution.getVariables().get("formUrl"));
+	}
+
+	/**
+	 * Returns the endpoint of Sync API.
+	 *
+	 * @return
+	 */
+	private String getSyncApplicationUrl() {
+		return this.reportUrl + "/sync-form-data";
```

Comment:
Instead of appending strings like this for URLs, can we use something like this.
`import java.nio.file.Paths;
import java.nio.file.Path;
class HelloWorld {
    public static void main(String[] args) {
        Path path = Paths.get("http://localhost", "application", "history");
        System.out.println(path); 
    }
}`